servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/PublishAndSubscribeOnSingles.java

```diff
@@ -40,38 +41,77 @@ private PublishAndSubscribeOnSingles() {
     }
 
     static <T> Single<T> publishAndSubscribeOn(Single<T> original, Executor executor) {
-        return original.executor() == executor ? original : new PublishAndSubscribeOn<>(executor, original);
+        return original.executor() == executor || immediate() == executor ?
```

Comment:
IIUC this will eventually go away as `original.executor()` will no longer exist?

Reply:
Correct, the `immediate() == executor` part might remain.